src/snowflake/snowpark/session.py

```diff
@@ -1568,6 +1568,65 @@ def query_tag(self, tag: str) -> None:
             self._conn.run_query("alter session unset query_tag")
         self._query_tag = tag
 
+    def append_query_tag(self, tag: Union[str, dict], seperator: str = ",") -> None:
+        """
+        Appends a tag to the current query tag.
+        When the input tag is a string the it will be appended to the query tag with a seperator.
+        When the input tag is a dictionary the query tag will be assumed to be a json encoded
+        dictionary. The query tag will be updated with the input tags keys and values.
+
+        Args:
+            tag: The tag to append to the current query tag. When the tag is a string the kj
+            seperator: When appending to a string tag this parameter will be used to seperated values.
+        Note:
+            Assigning a value via session.query_tag will remove any appended query tags.
+
+        Example::
+            >>> session.query_tag = "tag1"
+            >>> session.append_query_tag("tag2")
+            >>> print(session.query_tag)
+            tag1,tag2
+            >>> session.query_tag = "new_tag"
+            >>> print(session.query_tag)
+            new_tag
+
+        Example::
+            >>> session.sql("ALTER SESSION SET QUERY_TAG = 'tag1'").collect()
+            [Row(status='Statement executed successfully.')]
+            >>> session.append_query_tag("tag2")
+            >>> print(session.query_tag)
+            tag1,tag2
+
+        Example::
+            >>> session.query_tag = '{"key1": "value1"}'
+            >>> session.append_query_tag({"key2": "value2"})
+            >>> print(session.query_tag)
+            {"key1": "value1", "key2": "value2"}
+        """
+        if tag:
+            # Ensure local state matches remote
+            self._query_tag = self._conn.run_query("SHOW PARAMETERS LIKE 'QUERY_TAG'")[
+                "data"
+            ][0][1]
```

Comment:
What happens if `[0][1]` doesn't work anymore, maybe because server-side devs make a mistake, or we have a BCR but a user uses the first version of snowpark with this feature?
I think we should gracefully fail and provide a helpful error.